core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitorSpec.scala

```diff
@@ -830,6 +848,43 @@ class AvroUpsertMonitorSpec(_system: ActorSystem)
 
   }
 
+  it should "ack pubsub message if message is double delivered" in withTestDataApiServices { services =>
+    val timeout = 30000 milliseconds
+    val interval = 250 milliseconds
+    val importUuid = UUID.randomUUID()
+
+    setUpMockImportServiceToReturnStatus(services, ImportStatuses.Done)
+
+    val contents = makeOpsJsonString(100)
+
+    // Store upsert json file
+    Await.result(
+      googleStorage
+        .createBlob(bucketName, GcsBlobName(importUuid.toString), contents.getBytes())
+        .compile
+        .drain
+        .unsafeToFuture(),
+      Duration.apply(10, TimeUnit.SECONDS)
+    )
+
+    // acks should be empty at this point
+    eventually(Timeout(scaled(timeout)), Interval(scaled(interval))) {
+      services.gpsDAO.acks shouldBe empty
+    }
+
+    // Publish message on the request topic
+    services.gpsDAO.publishMessages(
+      importReadPubSubTopic,
+      List(MessageRequest(importUuid.toString, testAttributes(importUuid)))
+    )
+
+    // check that a pubsub message was acked.
+    eventually(Timeout(scaled(timeout)), Interval(scaled(interval))) {
+      services.gpsDAO.acks should not be empty
+    }
```

Comment:
Thinking out loud … I poked at this a bit, and really wanted to find a way to assert that it is acking the _second_ message - the duplicate one - but given the structure of `MockGooglePubSubDAO` I couldn't find a way to do that. I don't think it's worth refactoring a whole bunch of stuff to make that test work.